Allow any trim type to be specified in initialization file

seanmcleod opened the pull request:
Currently the `<trim>` element in an initialization file can only be used to specify `0 - no trim` or `1 - ground trim`. This pull requests adds support for all the currently supported trim types to be specified and maintains backwards compatibility.

`- trim (0 for no trim, 1 for ground trim, 'Longitudinal', 'Full', 'Ground', 'Pullup', 'Custom', 'Turn') `

bcoconni commented on src/initialization/FGInitialCondition.cpp:
```diff
@@ -979,6 +980,27 @@ bool FGInitialCondition::LoadLatitude(Element* position_el)
 
 //******************************************************************************
 
+void FGInitialCondition::SetTrimRequest(std::string& trim)
```
I think you need to use `const` for this code to compile under Travis CI
```c++
void FGInitialCondition::SetTrimRequest(const std::string& trim)
```

bcoconni commented on src/JSBSim.cpp:
```diff
@@ -454,10 +454,20 @@ int real_main(int argc, char* argv[])
     try {
+      // If PullUp requested then we need to pass in target Nlf
+      if (icTrimRequested == JSBSim::TrimMode::tPullup)
+        trimmer->SetTargetNlf(FDMExec->GetIC()->GetTargetNlfIC());
```
Why isn't that code in lines 462-464 included in `FGTrim::FGTrim` ?
I am pretty sure scripts will ignore the Nlf target otherwise.

seanmcleod replied:
I've moved the code to `FGTrim::FGTrim` which means the following will now also work which never worked previously in addition to pullup trim working from a `<trim>` element in the initialization file.

```xml
<!-- Pullup trim -->
<set name="simulation/do_simple_trim" value="4" />
```

seanmcleod commented:
I've fixed the gcc compiler error regarding the `string` argument and fixed the merge conflict. But I see now via Travis that there is another gcc only (compiles fine on AppVeyor) error now regarding a goto and local variable initialization etc. which I'll look into later today.

I'll also respond later to the question regarding `targetNlf`.
